stock_inventory_revaluation/models/stock_inventory_revaluation.py

```diff
@@ -0,0 +1,478 @@
+                                 "selected revaluation type is Price Change.",
+                            digits=dp.get_precision('Product Price'))
+
+    current_value = fields.Float('Current value',
+                                 help='Displays the current value of the '
+                                      'product.',
+                                 digits=dp.get_precision('Account'),
+                                 compute="_compute_calc_product_value",
+                                 readonly=True)
+
+    old_value = fields.Float('Old value',
+                             help='Displays the current value of the product.',
+                             digits=dp.get_precision('Account'),
+                             readonly=True)
+
+    new_value = fields.Float('Credit/Debit amount',
+                             help="Enter the amount you wish to credit or "
+                                  "debit from the current inventory value of "
+                                  "the item. Enter credit as a negative value."
+                                  "Relevant only if the selected revaluation "
+                                  "type is Inventory Credit/Debit.",
+                             digits=dp.get_precision('Account'))
+
+    qty_available = fields.Float(
+        'Quantity On Hand', compute='_compute_get_product_qty',
+        digits_compute=dp.get_precision('Product Unit of Measure'))
+
+    increase_account_id = fields.Many2one(
+        'account.account', 'Increase Account',
+        help="Define the G/L accounts to be used as the balancing account in "
+             "the transaction created by the revaluation. The Increase "
+             "Account is used when the inventory value is increased due to "
+             "the revaluation.")
+
+    decrease_account_id = fields.Many2one(
+        'account.account', 'Decrease Account',
+        help="Define the G/L accounts to be used as the balancing account in "
+             "the transaction created by the revaluation. The Decrease "
+             "Account is used when the inventory value is decreased.")
+
+    account_move_ids = fields.One2many(
+        comodel_name='account.move',
+        inverse_name='stock_inventory_revaluation_id',
+        readonly=True)
+
+    post_date = fields.Datetime(
+        'Posting Date', states={'done': [('readonly', True)]},
+        help="Date of actual processing")
+
+    reval_quant_ids = fields.One2many('stock.inventory.revaluation.quant',
+                                      'revaluation_id',
+                                      string='Revaluation line quants')
+
+    @api.multi
+    @api.depends("product_id", "product_id.standard_price")
+    def _compute_calc_current_cost(self):
+        for revaluation in self:
+            revaluation.current_cost = \
+                revaluation.product_id.standard_price
+
+    @api.multi
+    @api.constrains('product_id', 'company_id')
```

Comment:
@darshan-serpent can you remove this constraint? Is causing me trouble migrating from v8 to v9 with products that were stockable in the past but then changed to consumables